osquery/tables/system/windows/registry.cpp

```diff
@@ -116,114 +119,111 @@ inline void explodeRegistryPath(const std::string& path,
 }
 
 /// Microsoft helper function for getting the contents of a registry key
-void queryKey(const std::string& keyPath, QueryData& results) {
+Status queryKey(const std::string& keyPath, QueryData& results) {
   std::string hive;
   std::string key;
   explodeRegistryPath(keyPath, hive, key);
 
   if (kRegistryHives.count(hive) != 1) {
-    return;
+    return Status();
   }
 
-  HKEY hRegistryHandle;
-  auto ret = RegOpenKeyEx(kRegistryHives.at(hive),
-                          TEXT(key.c_str()),
-                          0,
-                          KEY_READ,
-                          &hRegistryHandle);
+  HKEY hkey;
+  auto ret = RegOpenKeyEx(
+      kRegistryHives.at(hive), TEXT(key.c_str()), 0, KEY_READ, &hkey);
+  reg_handle_t hRegistryHandle(hkey, closeRegHandle);
 
   if (ret != ERROR_SUCCESS) {
-    return;
+    return Status(GetLastError(), "Failed to open registry handle");
   }
 
   const DWORD maxKeyLength = 255;
   const DWORD maxValueName = 16383;
-  TCHAR achClass[MAX_PATH] = TEXT("");
-  DWORD cchClassName = MAX_PATH;
-  DWORD cSubKeys = 0;
-  DWORD cbMaxSubKey;
-  DWORD cchMaxClass;
+  DWORD cSubKeys;
   DWORD cValues;
   DWORD cchMaxValueName;
   DWORD cbMaxValueData;
-  DWORD cbSecurityDescriptor;
   DWORD retCode;
   FILETIME ftLastWriteTime;
-  retCode = RegQueryInfoKey(hRegistryHandle,
-                            achClass,
-                            &cchClassName,
+  retCode = RegQueryInfoKey(hRegistryHandle.get(),
+                            nullptr,
+                            nullptr,
                             nullptr,
                             &cSubKeys,
-                            &cbMaxSubKey,
-                            &cchMaxClass,
+                            nullptr,
+                            nullptr,
                             &cValues,
                             &cchMaxValueName,
                             &cbMaxValueData,
-                            &cbSecurityDescriptor,
+                            nullptr,
                             &ftLastWriteTime);
 
-  TCHAR achKey[maxKeyLength];
+  auto achKey = std::make_unique<TCHAR[]>(maxKeyLength);
```

Comment:
The arg type for achKey was TCHAR, but you're right that I can likely just skip the aliasing.